arangod/Cluster/ServerState.cpp

```diff
@@ -597,6 +598,28 @@ void ServerState::setId(std::string const& id) {
   _id = id;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief get the short server id
+////////////////////////////////////////////////////////////////////////////////
+
+uint32_t ServerState::getShortId() {
+  READ_LOCKER(readLocker, _lock);
+  return _shortId;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief set the short server id
+////////////////////////////////////////////////////////////////////////////////
+
+void ServerState::setShortId(uint32_t id) {
+  if (id == 0) {
+    return;
+  }
+
+  WRITE_LOCKER(writeLocker, _lock);
+  _shortId = id;
+}
```

Comment:
Make `_shortId` atomic to completely avoid the read/write lock in get/setShortId. In this case you can even use relaxed memory order for all operations.